### PR TITLE
Implemented first pass of uptime checks

### PR DESCRIPTION
**Pull request by StephenButtolph**

### CODE REVIEW

File: main/params.go [hunk 1]
```diff
@@ -203,9 +207,16 @@ func init() {
...
-	fs.Float64Var(&Config.StakerMsgPortion, "staker-msg-reserved", 0.2, "Reserve a portion of the chain message queue's space for stakers.")
+	fs.Float64Var(&Config.StakerMSGPortion, "staker-msg-reserved", 0.2, "Reserve a portion of the chain message queue's space for stakers.")
 	fs.Float64Var(&Config.StakerCPUPortion, "staker-cpu-reserved", 0.2, "Reserve a portion of the chain's CPU time for stakers.")
 
+	// Network Timeouts:
+	networkInitialTimeout := fs.Int64("network-initial-timeout", int64(10*time.Second), "Initial timeout value of the adaptive timeout manager.")
+	networkMinimumTimeout := fs.Int64("network-minimum-timeout", int64(500*time.Millisecond), "Minimum timeout value of the adaptive timeout manager.")
+	networkMaximumTimeout := fs.Int64("network-maximum-timeout", int64(10*time.Second), "Maximum timeout value of the adaptive timeout manager.")
+	fs.Float64Var(&Config.NetworkTimeoutMultiplier, "network-timeout-multiplier", 1.1, "Multiplier of the timeout after a failed request.")
+	networkTimeoutReduction := fs.Int64("network-timeout-reduction", int64(time.Millisecond), "Reduction of the timeout after a successful request.")
```
Comment (danlaine) [inline main/params.go:218]: `"Reduction of the timeout after a successful request, in milliseconds"`

File: node/node.go [hunk 2]
```diff
@@ -412,10 +413,31 @@ func (n *Node) initChainManager(avaxAssetID ids.ID) error {
...
+	go n.Log.RecoverAndPanic(timeoutManager.Dispatch)
+
+	n.Config.ConsensusRouter.Initialize(
+		n.Log,
+		&timeoutManager,
+		n.Config.ConsensusGossipFrequency,
+		n.Config.ConsensusShutdownTimeout,
+	)
+
+	n.chainManager = chains.New(
```
Comment (danlaine) [inline node/node.go:437]: We should probably just pass an arg struct rather than passing args like this

File: main/params.go (hunk 1 above)
Comment (StephenButtolph) [reply main/params.go]: nanoseconds* but I added the units to the description.

File: snow/networking/router/chain_router_test.go [hunk 3]
```diff
@@ -70,7 +78,15 @@ func TestShutdown(t *testing.T) {
 func TestShutdownTimesOut(t *testing.T) {
 	tm := timeout.Manager{}
 	// Ensure that the MultiPut request does not timeout
-	tm.Initialize("", prometheus.NewRegistry())
+	tm.Initialize(
```
Comment (danlaine) [inline snow/networking/router/chain_router_test.go:81]: We should probably just pass an arg struct rather than passing args like this

Comment (StephenButtolph) [reply snow/networking/router/chain_router_test.go]: Added a config.

File: node/node.go (hunk 2 above)
Comment (StephenButtolph) [reply node/node.go]: Agreed.

**Pull request merged**